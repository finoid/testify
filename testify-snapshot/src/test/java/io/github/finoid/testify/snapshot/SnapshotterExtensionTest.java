package io.github.finoid.testify.snapshot;

import org.junit.jupiter.api.Test;
import org.junit.jupiter.api.extension.ExtendWith;

@ExtendWith(SnapshotterExtension.class)
class SnapshotterExtensionTest {
    @Test
    void givenSnapshotter_whenUserSnapshottedAsJson_thenSnapshotCreated(Snapshotter snapshotter) {
        snapshotter.json()
            .snapshot(new User(1, "John"));
    }

    @Test
    void givenSnapshotter_whenUserSnapshottedAsPlain_thenSnapshotCreated(Snapshotter snapshotter) {
        snapshotter.plain()
            .snapshot(new User(1, "John"));
    }

    @Test
    void givenSnapshotter_whenUserSnapshottedAsBase64_thenSnapshotCreated(Snapshotter snapshotter) {
        snapshotter.base64()
            .snapshot(new User(1, "John"));
    }

    record User(int id, String name) {}
}