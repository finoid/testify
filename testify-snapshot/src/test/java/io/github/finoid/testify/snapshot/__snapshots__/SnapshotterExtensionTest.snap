io.github.finoid.testify.snapshot.SnapshotterExtensionTest.givenSnapshotter_whenUserSnapshottedAsBase64_thenSnapshotCreated=[
VXNlcltpZD0xLCBuYW1lPUpvaG5d
]


io.github.finoid.testify.snapshot.SnapshotterExtensionTest.givenSnapshotter_whenUserSnapshottedAsJson_thenSnapshotCreated=[
  {
    "id" : 1,
    "name" : "John"
  }
]


io.github.finoid.testify.snapshot.SnapshotterExtensionTest.givenSnapshotter_whenUserSnapshottedAsPlain_thenSnapshotCreated=[
User[id=1, name=John]
]